pkg/containerutil/cp_linux.go

```diff
@@ -247,3 +275,46 @@ func mountSnapshotForContainer(ctx context.Context, client *containerd.Client, c
 	}
 	return tempDir, cleanup, nil
 }
+
+func getContainerMountInfo(ctx context.Context, con containerd.Container, containerPath string) (string, string, error) {
+	filePath := filepath.Clean(containerPath)
+	spec, err := con.Spec(ctx)
+	if err != nil {
+		return "", "", err
+	}
+	if spec.Root.Readonly {
+		return "", "", fmt.Errorf("container rootfs: %s is marked read-only", spec.Root.Path)
+	}
+
+	for _, mount := range spec.Mounts {
+		if isParentDir(filePath, mount.Destination) {
```

Comment:
We should manage the priority of mounts with the same destination. For example if a user do `-v /dir:/tmp:ro` we should avoid copying files into `/tmp` which is readonly. But it looks like the first mount is prioritized here so it can cause unexpected behaviour in the situation like the above?

Reply:
`isParentDir` checks is user is trying to copy to subdirectory of a mounted volume. Eg `isParentDir("/tmp/some_nested_dir", "/tmp")` will return `true `and if `/tmp` is a `ro`, it'll throw an error. 

- If a user creates multiple volumes with the same mount point `sudo nerdctl run -v myvol:/tmp:ro -v myvol2:/tmp alpine`, it's anyway ambiguous to whether /tmp should be read-only or not irrespective of the order. Although nerdctl allows mounting multiple  volumes to the same mount point(/tmp in this example), docker throws an error with message `Duplicate mount point: /tmp`. Should this perhaps be fixed in the run command to not allow same mount point?

Reply:
> Should this perhaps be fixed in the run command to not allow same mount point?

I think yes it should. Even with this limitation, I believe user still can specify a single `-v /dir:/tmp:ro` which cause duplication on /tmp for the default tmpfs mount and the user-specified mount so the duplication management is still needed here.

Reply:
From the [runtime spec](https://github.com/opencontainers/runtime-spec/blob/main/specs-go/config.go#L18) `spec.Mounts` contains additional mounts on top of root so the default tmps mount won't show up in this list.

> single -v /dir:/tmp:ro which cause duplication on /tmp for the default tmpfs mount

I added a [test](https://github.com/containerd/nerdctl/pull/2413/files#diff-fc1f66854a3bb061fe38564a668b1b07c4be8fe9f404e8248b80ab2edad6e22fR156) for this scenario to confirm correct behaviour of not allowing to write to /tmp. Maybe I missed something here, can you help define a test case to describe the scenario for duplication and allowing write to a read-only volume?